src/consensus/chainsel.md

```diff
@@ -0,0 +1,259 @@
+completely determined some positive number of slots ago on that header's chain.
+
+## The `k` security parameter
+
+Both Ouroboros Classic and Ouroboros Praos are based on a chain selection rule
+that imposes a maximum rollback condition: alternative chains to a node's
+current chain that fork off more than a certain number of blocks ago are never
+considered for adoption. This limit is known as the security parameter, and is
+usually denoted by `k`, which on mainnet is currently set to 2160
+blocks. Analysis from research has shown this parameter ensures nodes will not
+diverge more than `k` blocks under honest circumstances with sufficiently high
+probability.
+
+Ouroboros BFT does not impose a maximum rollback, but adding such a requirement
+does not change the protocol in any fundamental way.
+
+The Consensus layer must evaluate the validity of any candidate chain. To
+evaluate the validity of a particular header, a Chain State at the predecessor
+header is required. To evaluate the validity of a particular block, a Ledger
+State at the predecessor block is required.
+
+> [!TIP]
+>
+> Along the selected chain, the combined Ledger and Chain states should be kept
+> in memory for all the `k` volatile blocks (and for the immutable tip, so `k+1`
+> states). This way, evaluation of candidates can be performed in a timely
+> manner.
+
+Making use of this parameter, the chain can be subdivided into the Immutable and
+Volatile parts of the chain:
+
+```mermaid
+flowchart RL
+    subgraph "Volatile chain";
+      subgraph "current selection";
+        Tip -->|...| Km1["(k-1)-th Block"]
+        Km1 --> K["k-th Block"]
+      end
+
+      O1["Block"] --> O2["Block"]
+      O3["Block"] --> K
+
+      O4["Block"]
+    end
+
+    subgraph "Immutable chain";
+     Kp1["(k+1)-th Block"] --> Kp2["(k+2)-th Block"]
+     Kp2["(k+2)-th Block"] --> |...| A[Genesis]
+    end
+    K --> Kp1
+```
+
+The Immutable part can be persisted in the storage as Ouroboros guarantees it
+will not change, whereas the selection in the Volatile part might change as new
+blocks arrive, creating new candidates.
+
+## The Chain Selection Rule
+
+The specific rule for determining the best chain is dictated by the active era's
+protocol.
+
+### Ouroboros Classic
+
+In Ouroboros Classic, the protocol defines the function ${\tt
+maxvalid}_S(\mathcal{C}, \mathbb{C})$ for choosing the best chain considering
+the current selected chain $\mathcal{C}$ and the set of valid chains
+$\mathbb{C}$ available in the network. The rule states that the chosen candidate
+must be the longest one, breaking ties in favor of $\mathcal{C}$.
+
+See the definition of ${\tt maxvalid}$ in section 4.1 in the [paper][classic].
+
+### Ouroboros BFT and PBFT
+
+In Ouroboros BFT, the protocol declares that a node should replace the local
+chain $B_{0}B_{1}\ldots{}B_{l}$ with a different chain
+$B_{0}{B'}_{1}\ldots{B'}_{s}$ if $s > l$, therefore if the candidate is longer
```

Comment:
The LaTeX rendering of this formula does not seem to work in the GH preview. Maybe it does in the mdbook?

Reply:
It does work in mdbook 

![image](https://github.com/user-attachments/assets/bfbb5fee-2c21-4b97-b680-dbc4bb667beb)